src/main/java/org/folio/circulation/domain/LocationRepository.java

```diff
@@ -103,4 +108,34 @@ public static LocationRepository using(Clients clients) {
       .fetch(location.getInstitutionId())
       .thenApply(r -> r.map(location::withInstitutionRepresentation));
   }
+
+  private CompletableFuture<Result<MultipleRecords<Location>>> loadLibrariesForLocations(
+          Result<MultipleRecords<Location>> multipleRecordsResult) {
+
+    return multipleRecordsResult.after(locations -> getLibraries(locations.getRecords())
```

Comment:
This can be simplified by using the `mapRecords` method of `MultipleRecords`

```
    return multipleRecordsResult.combineAfter(
      locations -> getLibraries(locations.getRecords()), (locations, libraries) ->
        locations.mapRecords(location -> location.withLibraryRepresentation(
          libraries.getOrDefault(location.getLibraryId(), null))));
```

I would likely extract a method for the matching up, but wanted to include it inline here for simplicity of sharing the example.